File: examples/hello.py
"""Sample script for the Run button: shows which interpreter actually ran it."""

import sys
from pathlib import Path


def main() -> None:
    print("Hello from CleeCode")
    print(f"interpreter: {Path(sys.executable)}")
    print(f"python:      {sys.version.split()[0]}")
    total = sum(n * n for n in range(1, 11))
    print(f"sum of squares 1..10 = {total}")


if __name__ == "__main__":
    main()

File: examples/hello.m
% Sample script for the Run button. `octave --no-gui --persist` keeps the interpreter (and any
% plot windows) alive after it finishes, so the session stays inspectable.
printf("Hello from CleeCode\n");
printf("octave:  %s\n", version());
A = [1 2; 3 4];
printf("det([1 2; 3 4]) = %g\n", det(A));
printf("mean of 1..10   = %g\n", mean(1:10));

figure(1);
plot(A(1,:),A(2,:));
grid minor;
